interfaces/builtin/polkit.go

```diff
@@ -145,11 +151,168 @@ func (iface *polkitInterface) PolkitConnectedPlug(spec *polkit.Specification, pl
 	return nil
 }
 
-func (iface *polkitInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+type polkitInstallRule struct {
+	Name, Sha3_384 string
+}
+
+func (iface *polkitInterface) parseAndValidateInstallRules(attribs interfaces.Attrer) ([]polkitInstallRule, error) {
+	var ruleEntries []map[string]string
+	if err := attribs.Attr("install-rules", &ruleEntries); err != nil {
+		return nil, err
+	}
+	if len(ruleEntries) == 0 {
+		return nil, fmt.Errorf("\"install-rules\" must have at least one entry")
+	}
+	rules := make([]polkitInstallRule, len(ruleEntries))
+	for i, ruleEntry := range ruleEntries {
+		rule := polkitInstallRule{}
+		for key, val := range ruleEntry {
+			switch key {
+			case "name":
+				if err := validate.ValidateRuleFileName(val); err != nil {
+					return nil, err
+				}
+				rule.Name = val
+			case "sha3-384":
+				rule.Sha3_384 = val
+			default:
+				return nil, fmt.Errorf("unexpected key %q for \"install-rules\" entry", key)
```

Comment:
```suggestion
				return nil, fmt.Errorf(`unexpected key %q for "install-rules" entry`, key)
```
and maybe in other places when using key names with quotation marks